include/pybind11/pybind11.h

```diff
@@ -1206,10 +1206,13 @@ class generic_type : public object {
         if (rec.bases.size() > 1 || rec.multiple_inheritance) {
             mark_parents_nonsimple(tinfo->type);
             tinfo->simple_ancestors = false;
+            tinfo->simple_type = false;
         }
         else if (rec.bases.size() == 1) {
             auto parent_tinfo = get_type_info((PyTypeObject *) rec.bases[0].ptr());
             tinfo->simple_ancestors = parent_tinfo->simple_ancestors;
+            // a child of a non-simple type can never be a simple type
+            tinfo->simple_type = parent_tinfo->simple_type;
```

Comment:
Nit: Technically should be "&&" because non-simple parent implies non-simple type, but simple parent may not imply simple type by it's current definition.